### PR TITLE
Add support for Terraform v1.x

### PR DESCRIPTION
**Pull request by soerenmartius**
- feat: add support for Terraform v1.x
- chore: prepare v0.5.0 release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
 - Add terraform `v1.x` support
+- Add `aws_route_table` as dependencies to `aws_route` resources
```
Comment (mariux) [inline CHANGELOG.md:15]: not change din this commit. please always update CHANGELOG in seperate commit to easier resolve conflicts

File: private_routing.tf [hunk 2]
```diff
@@ -135,5 +135,6 @@ resource "aws_route" "nat_gateway" {
   depends_on = [
     var.module_depends_on,
     aws_nat_gateway.nat_gateway,
+    aws_route_table.private
```
Comment (mariux) [inline private_routing.tf:133]: why? a nat gateway does not need a route table to exist before it is created!?

File: public_routing.tf [hunk 3]
```diff
@@ -72,5 +72,9 @@ resource "aws_route" "internet_gateway" {
     create = "10m"
   }
 
-  depends_on = [var.module_depends_on]
+  depends_on = [
+    var.module_depends_on,
+    aws_internet_gateway.internet_gateway,
+    aws_route_table.public
```
Comment (mariux) [inline public_routing.tf:73]: why? they are already known as dependencies by terraform as they are used in the resource

**Review by mariux — CHANGES_REQUESTED** [review]

**Comment by mariux** [comment]
ping @soerenmartius any progress on this PR?

File: private_routing.tf (hunk 2 above)
Comment (soerenmartius) [reply private_routing.tf:133]: > ping @soerenmartius any progress on this PR?

no time at the moment - will convert back to a draft and address once we plan this accordingly. The dependencies didn't resolve properly btw that's why I added them to `depends_on`

**Comment by JimFawkes** [comment]
Hey ho,
just want to flag interest for this change, it would be awesome to see support for TF v1 released for this module.

Thanks by the way for providing these modules as OSS

**Comment by soerenmartius** [comment]
> Hey ho,
> just want to flag interest for this change, it would be awesome to see support for TF v1 released for this module.
> 
> Thanks by the way for providing these modules and the great work you do.

Hi @JimFawkes ,

Thank you for the flowers :) I will see to get this finished during the weekend (finally) so we can get it released at latest by the beginning of next week.

Thanks for using our modules and have a nice weekend!

**Comment by soerenmartius** [comment]
@JimFawkes @mariux the PR is r4r!

Seems like our testing AWS reached some quotas. Will take a detailed look tomorrow so that the tests can run properly.

https://github.com/mineiros-io/terraform-aws-vpc/pull/35/checks?check_run_id=3513839623#step:3:1383

**Comment by soerenmartius** [comment]
> @JimFawkes @mariux the PR is r4r!
> 
> Seems like our testing AWS reached some quotas. Will take a detailed look tomorrow so that the tests can run properly.
> 
> https://github.com/mineiros-io/terraform-aws-vpc/pull/35/checks?check_run_id=3513839623#step:3:1383

@mariux cleaned up the testing account - tests are passing now

**Review by mariux — APPROVED** [review]
lgtm

**Comment by soerenmartius** [comment]
Hi @JimFawkes , we just released [v0.5.0](https://github.com/mineiros-io/terraform-aws-vpc/releases/tag/v0.5.0) now!

**Comment by JimFawkes** [comment]
@soerenmartius That was fast! Thank you very much! I'll give it a spin now.

**Pull request merged**